rest_framework/renderers.py

```diff
@@ -567,6 +567,8 @@ def get_raw_data_form(self, data, view, method, request):
                     if isinstance(field, serializers.HiddenField):
                         data.pop(name, None)
                 content = renderer.render(data, accepted, context)
+                # Renders returns bytes, but CharField expects a str.
+                content = content.decode('utf-8')
```

Comment:
I still don't get why this is required. I'm afraid we create a regression here.

Reply:
If you run with the -b flag we get a BytesWarning from the (Django) widget rendering the value here. The conversion to str resolves this. 

There’s no issue on Python 2.7. You decode bytes into Unicode string.